test/unit/specs/components/Identity/index.spec.js

```diff
@@ -59,5 +61,61 @@ describe("Identity", () => {
         expect(getEcid()).toBeUndefined();
       });
     });
+
+    describe("onResponse", () => {
+      let cookieJar;
+      let response;
+      beforeEach(() => {
+        cookieJar = {
+          get(key) {
+            return key === "ECID" ? "ABC" : null;
+          },
+          set: jasmine.createSpy()
+        };
+        response = {
+          getPayloadsByType: jasmine
+            .createSpy()
+            .and.returnValue([{ id: "ABC" }]),
+          toJSON: jasmine.createSpy()
+        };
+
+        identity = createIdentity({
+          config: {
+            reactorRegisterGetEcid
+          },
+          cookieJar
+        });
+        identity.lifecycle.onComponentsRegistered({
+          optIn: {
+            whenOptedIn() {
+              return Promise.resolve();
+            }
+          }
+        });
+      });
+      it("should get called with an object with a property named response in it", () => {
+        identity.lifecycle.onResponse({ response });
+        return flushPromiseChains().then(() => {
+          expect(response.getPayloadsByType).toHaveBeenCalledWith(
+            "identity:persist"
+          );
+          expect(cookieJar.set).toHaveBeenCalledWith(
+            EXPERIENCE_CLOUD_ID,
+            "ABC"
+          );
+          expect(response.getPayloadsByType).toHaveBeenCalledWith(
+            "identity:exchange"
+          );
+        });
+      });
+      it("should not do anything if the wrong object is passed", () => {
```

Comment:
A better test might be for when getPayloadsByType("identity:persist") returns [].

Reply:
I agree with this. Components shouldn't have to code for, or test, the possible case where `onResponse` is called with an invalid object because we own all the code that could potentially call `lifecycle.onResponse`, and that code should have its own tests to ensure it's passing the right object.